src/controllers/dynakube/dynakube_controller.go

```diff
@@ -174,183 +170,172 @@ func (controller *DynakubeController) reconcileIstio(dynakube *dynatracev1beta1.
 		}
 	} else {
 		if err := dkMapper.UnmapFromDynaKube(); err != nil {
-			log.Error(err, "could not unmap dynakube from namespace")
-			return
+			log.Info("could not unmap dynakube from namespace")
+			return err
 		}
-		err = endpointSecretGenerator.RemoveEndpointSecrets(ctx, dkState.Instance)
-		if dkState.Error(err) {
-			return
+		err = endpointSecretGenerator.RemoveEndpointSecrets(ctx, dynakube)
+		if err != nil {
+			log.Info("could not remove data-ingest secret")
+			return err
 		}
 	}
-
-	upd = controller.determineDynaKubePhase(dkState.Instance)
-	dkState.Update(upd, "dynakube phase changed")
+	return nil
 }
 
-func (controller *DynakubeController) reconcileOneAgent(ctx context.Context, dkState *status.DynakubeState, dtc dtclient.Client) (err error) {
-	if dkState.Instance.HostMonitoringMode() {
-		upd, err := oneagent.NewOneAgentReconciler(
-			controller.client, controller.apiReader, controller.scheme, dkState.Instance, daemonset.DeploymentTypeHostMonitoring,
-		).Reconcile(ctx, dkState)
-		if dkState.Error(err) {
-			return nil
+func (controller *DynakubeController) reconcileOneAgent(ctx context.Context, dynakube *dynatracev1beta1.DynaKube) (err error) {
+	if dynakube.HostMonitoringMode() {
+		err = oneagent.NewOneAgentReconciler(
+			controller.client, controller.apiReader, controller.scheme, daemonset.DeploymentTypeHostMonitoring,
+		).Reconcile(ctx, dynakube)
+		if err != nil {
+			return err
 		}
-		dkState.Update(upd, "host monitoring reconciled")
-	} else if dkState.Instance.CloudNativeFullstackMode() {
-		upd, err := oneagent.NewOneAgentReconciler(
-			controller.client, controller.apiReader, controller.scheme, dkState.Instance, daemonset.DeploymentTypeCloudNative,
-		).Reconcile(ctx, dkState)
-		if dkState.Error(err) {
-			return nil
+		log.Info("reconciled host-monitoring")
+	} else if dynakube.CloudNativeFullstackMode() {
+		err = oneagent.NewOneAgentReconciler(
+			controller.client, controller.apiReader, controller.scheme, daemonset.DeploymentTypeCloudNative,
+		).Reconcile(ctx, dynakube)
+		if err != nil {
+			return err
 		}
-		dkState.Update(upd, "cloud native fullstack monitoring reconciled")
-	} else if dkState.Instance.ClassicFullStackMode() {
-		upd, err := oneagent.NewOneAgentReconciler(
-			controller.client, controller.apiReader, controller.scheme, dkState.Instance, daemonset.DeploymentTypeFullStack,
-		).Reconcile(ctx, dkState)
-		if dkState.Error(err) {
-			return nil
+		log.Info("reconciled cloud-native")
+	} else if dynakube.ClassicFullStackMode() {
+		err = oneagent.NewOneAgentReconciler(
+			controller.client, controller.apiReader, controller.scheme, daemonset.DeploymentTypeFullStack,
+		).Reconcile(ctx, dynakube)
+		if err != nil {
+			return err
 		}
-		dkState.Update(upd, "classic fullstack reconciled")
+		log.Info("reconciled classic-fullstack")
 	} else {
-		controller.removeOneAgentDaemonSet(dkState)
+		controller.removeOneAgentDaemonSet(ctx, dynakube)
```

Comment:
do we have a log somewhere that we successfully removed the daemonset? also seems like we are ignoring any errors that may happen in this method